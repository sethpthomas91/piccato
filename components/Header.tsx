import styles from '../styles/header.module.css';
import Router from 'next/router';
import Image from 'next/image';
import Head from 'next/head'

const Header = () => {
  return (
    <nav className={styles.header}>
      <Head>
        <title>Pic-Cat-O</title>
        <meta name="description" content="where people go to get catified" />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <h1 className={styles.title} 
        data-testid="titlePresser"
        onClick={() => Router.push('/')}>
          PIC-CAT-TO
      </h1>
      <a onClick={() => Router.push('/create/')}>
        <Image src='/create_icon.jpg' height='80px' width='80px'/>
      </a>
    </nav>
  );
}

export default Header
