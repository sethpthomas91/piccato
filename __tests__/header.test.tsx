import renderer from 'react-test-renderer';
import Header from '../components/Header';
import '@testing-library/jest-dom'

describe ('Header', () => {

    it('renders correctly', () => {
      const tree = renderer
        .create(<Header/>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
})
