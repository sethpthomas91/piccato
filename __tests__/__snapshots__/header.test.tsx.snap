// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="header"
>
  <h1
    className="title"
    data-testid="titlePresser"
    onClick={[Function]}
  >
    PIC-CAT-TO
  </h1>
  <a
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=%2Fcreate_icon.jpg&w=256&q=75"
          srcSet="/_next/image?url=%2Fcreate_icon.jpg&w=96&q=75 1x, /_next/image?url=%2Fcreate_icon.jpg&w=256&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </a>
</nav>
`;
